client/go/internal/vespa/document/dispatcher.go

```diff
@@ -84,6 +82,8 @@ func (d *Dispatcher) sendDocumentIn(group *documentGroup) {
 	op.attempts++
 	result := d.feeder.Send(op.document)
 	d.results <- result
+	d.releaseSlot()
+	group.mu.Unlock()
```

Comment:
Slots in the Java implementation are per-ID, so a retry uses the same slot as the failed operation, and makes in unnecessary to acquire a new lease for both retries and for operations which are blocked due to an inflight operation with the same ID. 
Is this change because you'd otherwise risk blocking your thread trying to acquire a slot?

Reply:
> Slots in the Java implementation are per-ID

Right, I can look into doing that in a later PR.

>  Is this change because you'd otherwise risk blocking your thread trying to acquire a slot?

No, it's to prevent a deadlock on retry since we previously held the lock while enqueuing the retry (this attempts to take the same lock and locks are not re-entrant).